### PR TITLE
Get accounts SSO extension handling with shared device support

### PR DESCRIPTION
**Pull request by oldalton**
This PR includes:
1. Implementation of the getAccounts SSO extension request.
2. MSIDGetAccountsBrokerResponse fixes that are needed for nr 1. Previous implementation was using an array inside the dictionary, which crashed underlying SSO extension implementation inside CFNetwork. New implementation encodes account array into JSON instead. 
3. Allow saving "current account" into account metadata to track current account in shared device scenarios.

### CODE REVIEW

File: IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetAccountsResponse.m [hunk 1]
```diff
@@ -46,12 +48,29 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)er
...
+        {
+            self.accounts = @[];
+            self.success = YES;
+            return self;
+        }
+        
+        NSData *jsonData = [accountsString dataUsingEncoding:NSUTF8StringEncoding];
+        NSArray *accountsJson = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:error];
+                  
+        if (!accountsJson)
```
Comment (jasoncoolmax) [inline IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetAccountsResponse.m]: Do we want to make sure `accountsJson` is an array? :)

File: IdentityCore/src/cache/metadata/accountMetadata/MSIDAccountMetadataCacheItem.m [hunk 2]
```diff
@@ -96,6 +97,14 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json
     _accountMetadataMap = [NSMutableDictionary new];
     _clientId = [json msidStringObjectForKey:MSID_CLIENT_ID_CACHE_KEY];
     
+    NSString *principalHomeAccountId = [json msidStringObjectForKey:MSID_PRINCIPAL_HOME_ACCOUNT_ID_CACHE_KEY];
+    NSString *principalDisplayableId = [json msidStringObjectForKey:MSID_PRINCIPAL_DISPLAYABLE_ID_CACHE_KEY];
+    
+    if (principalHomeAccountId ||principalDisplayableId)
```
Comment (jasoncoolmax) [inline IdentityCore/src/cache/metadata/accountMetadata/MSIDAccountMetadataCacheItem.m]: nit: space after `||`

File: IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetAccountsResponse.m (hunk 1 above)
Comment (oldalton) [reply IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetAccountsResponse.m]: Addressed :)

File: IdentityCore/src/cache/metadata/accountMetadata/MSIDAccountMetadataCacheItem.m (hunk 2 above)
Comment (oldalton) [reply IdentityCore/src/cache/metadata/accountMetadata/MSIDAccountMetadataCacheItem.m]: Addressed :)

**Pull request merged**